### ISSUE TITLE
S3 Backend not use profile if you have AWS env variables set

### ISSUE THREAD

**Issue by clsonel-endava** [hashicorp/terraform#32577]
### Terraform Version

```shell
Terraform v1.3.7
on linux_amd64
```

### Terraform Configuration Files

```terraform
terraform {
  backend "s3" {
    encrypt = true
    profile = "my_profile"
    bucket  = "my_bucket"
    key     = "my_tfstate.tfstate"
    region  = "eu-west-1"
  }
  required_version = "1.3.7"
}
```

### Debug Output

https://gist.github.com/clsonel-endava/7fbbc29296e0fe8ad2626cf9b02d0e08

### Expected Behavior

Initializing the backend...
2023-01-25T17:01:42.671+0200 [INFO]  Attempting to use session-derived credentials
2023-01-25T17:01:48.292+0200 [INFO]  Successfully derived credentials from session
2023-01-25T17:01:48.292+0200 [INFO]  AWS Auth provider used: "AssumeRoleProvider"

### Actual Behavior

Initializing the backend...
2023-01-25T15:02:59.541Z [INFO]  AWS Auth provider used: "EnvProvider"

### Steps to Reproduce

`terraform init`

### Additional Context

profile is simply ignored

**Comment by crw** [comment]
Thanks for the report, I put it on the AWS Provider team's triage queue.

**Comment by gdavison** [comment]
This will be resolved when #30443 is complete. It is an active project this quarter.

**Comment by gdavison** [comment]
This will be addressed in Terraform v1.6. If you still see this error when v1.6 is released (or you try any of the v1.6 beta releases), please open a new issue
